# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLTemplateElement.h
# Commit: f9cf0740eef2
# Full Hash: f9cf0740eef2fcae3e2f93afe49df1569bd89c03
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 37e2a10a9022 -r f9cf0740eef2 dom/html/HTMLTemplateElement.h
--- a/dom/html/HTMLTemplateElement.h	Tue May 14 16:43:27 2024 +0000
+++ b/dom/html/HTMLTemplateElement.h	Tue May 14 16:43:28 2024 +0000
@@ -65,6 +65,13 @@
     SetHTMLBoolAttr(nsGkAtoms::shadowrootclonable, aValue, aRv);
   }
 
+  bool ShadowRootSerializable() const {
+    return GetBoolAttr(nsGkAtoms::shadowrootserializable);
+  }
+  void SetShadowRootSerializable(bool aValue, ErrorResult& aRv) {
+    SetHTMLBoolAttr(nsGkAtoms::shadowrootserializable, aValue, aRv);
+  }
+
   MOZ_CAN_RUN_SCRIPT
   void SetHTMLUnsafe(const nsAString& aHTML) final;
 